File: src/models/Block.ts
export default class Block {

    private index: number;
    private hash: string;
    private previousHash: string;
    private timestamp: number;
    private data: string;

    constructor(index: number, hash: string, previousHash: string, timestamp: number, data: string) {
        this.index = index;
        this.hash = hash;
        this.previousHash = previousHash;
        this.timestamp = timestamp;
        this.data = data;
    }

    getIndex = (): number => {
        return this.index;
    }

    getHash = (): string => {
        return this.hash;
    }

    getPreviousHash = (): string => {
        return this.previousHash;
    }

    getTimestamp = (): number => {
        return this.timestamp;
    }

    getData = (): string => {
        return this.data;
    }

    calculateHash = (index: number, previousHash: number, timestamp: number, data: string) => {
        
    }
}

File: src/main/java/com/dummycoin/dummycoin/model/Block.java
package com.dummycoin.dummycoin.model;

import com.dummycoin.dummycoin.util.StringUtil;

import java.util.Date;

public class Block {
    private String hash;
    private String previousHash;
    private String data;
    private long timeStamp;

    public Block(String data, String previousHash) {
        this.data = data;
        this.previousHash = previousHash;
        this.timeStamp = new Date().getTime();
        this.hash = this.calculateHash();
    }

    public String getHash() {
        return hash;
    }

    public String getPreviousHash() {
        return previousHash;
    }

    public String getData() {
        return data;
    }

    public long getTimeStamp() {
        return timeStamp;
    }

    public String calculateHash() {
        return StringUtil.applySHA256(this.previousHash + Long.toString(this.timeStamp) + this.data);
    }
}
